### ISSUE TITLE
Edit button overlaps content

### ISSUE THREAD

**Issue by IngoBelka** [nextcloud/notes-android#2272]
### This issue respects the following points:

- [X] I have read the [FAQ](https://github.com/nextcloud/notes-android/blob/master/FAQ.md)
- [X] I have searched for existing issues
- [X] The issue affects the Notes Android app, neither the [Notes server app](https://github.com/nextcloud/notes/issues/) nor the [Nextcloud Android app](https://github.com/nextcloud/android/issues).

### Describe the bug

1. start app
2. open a note
3. scroll down to the end
4. Edit button overlaps the last lines of the note

### Expected behavior

no overlapping of content
bring back the edit icon in menu line

### Notes Android version

4.3 f-droid

### Notes server version

4.10

### Nextcloud Android version

3.29.1 f-droid

### Nextcloud version

27.1.8

### Device

Huawei media pad

### Android Version

9.1

### App Store

- [ ] Google Play Store
- [X] F-Droid
- [ ] Huawei App Gallery

**Comment by Buntbart** [comment]
Please, this annoying button shouldn't be too hard to fix. The function of this floating button would be clear enough if it only showed the pen icon and nothing more.
It would be better to offer an option in the settings, which editor (plain/rich) should be preferred.

**Comment by pbek** [comment]
> Please, this annoying button shouldn't be too hard to fix.

Did you just volunteer? 😊

**Comment by Buntbart** [comment]
If I could program, I would have already offered. But since it's only a very small change, I strongly suspect that it shouldn't take much more effort for people who can do that than replying to people who post error messages here.

**Comment by Buntbart** [comment]
No, I have not. So you say, it needs big efforts to change the label of a button? Or do you want to say that you don't want to be bothered by users?

**Comment by pbek** [comment]
> How much effort does it take to change the label of a button without changing it's functionality?

Do you think that fixes the issue?

**Comment by papjul** [comment]
It would be helpful to tag issues open to contributions as such, and share what the project maintainers expect as a fix for this issue because there are design concerns behind.

There are many solutions I can think of:
1. Get back to old behavior: no FAB button => fine for me, but I'm not sure there is another way to access the advanced editor for people who like this, right?
2. Add a safe zone: no text behind the FAB button, the textarea stops above the button
3. Move the button as an option of the 3 dots menu
4. Move the option as a global preference

Depending on your choice (or any other not listed), the fix can be very easy or requires more knowledge of the code for someone outside the project. At first sight, someone familiar with the code shouldn't be having too much difficulty applying one of the option I suggested (but I get you are volunteers and you might have other priorities).

**Comment by Buntbart** [comment]
> Do you think that fixes the issue?

For my needs, this would solve the problem.

**Comment by looowizz** [comment]
Imo [tabs](https://m3.material.io/components/tabs/overview) could be well-suited: a preview, plain edit and rich edit tab?

**Comment by joshtrichards** [comment]
#1904, which was added in response to #1819, may help some of you here.

**Comment by fm-sys** [comment]
This bug was annoying enough that I filed a Pull Request (#2672). Let's keep fingers cross that the PR gets accepted, and a new version is released soon.

**Comment by Buntbart** [comment]
Yesterday, I updated to 4.4.0 (F-Droid) and unfortunately, now I can no longer edit at the bottom of the text because it is covered by the keyboard.

**Comment by tretkowski** [comment]
Please reopen, this is still (or again) an issue in 33.0.0.

**Comment by daniele-verducci** [comment]
Hey, @tretkowski , can you check if it's still there in 34.0.1 and eventually provide more info on how to reproduce it? I tested it but couldn't reproduce: the Floating Action Button does disappear while scrolling and the text has an end padding, so it's always visible on my device. Here's how I tested:

https://github.com/user-attachments/assets/f03f2549-c978-451d-b444-e714336c4e41

Any advice on how to reproduce it?

**Comment by fm-sys** [comment]
It's still there, you can reproduce it with the following settings:

<img width="1080" height="772" alt="Image" src="https://github.com/user-attachments/assets/c9585d92-ed55-482e-9332-5767a499bc1d" />
<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/4895fa49-fcaf-4c0b-8302-94ba97ffc27b" />

**Comment by fm-sys** [comment]
Btw, that's for taking a look!

Since the layout got messed up again just 1 month after my initial PR (#2672), I didn't bothered with creating a second one to fix things again... I'll keep fingers crossed that things will last longer than last time 😇

As already mentioned in https://github.com/nextcloud/notes-android/pull/2672#issuecomment-3127773737, the bug was introduced by https://github.com/nextcloud/notes-android/commit/caa792bc75dcdfcb6181fd0503c33e7e60fa33f5

**Comment by papjul** [comment]
I don't think you need any particular setting.

Paste some long text or some Lorem Ipsum:
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/f89e4084-e468-45ce-b73f-23571935b330" />

Delete the last few characters; this will trigger the vertical scroll following the deletion of the last line, and the floating button will cover all the text from the previous line that you're trying to edit:
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/d463c992-2037-49a9-83a6-2daf13dcd29f" />

This seems like a marginal case, but deleting a few characters to just change the wording of some sentence happen more often than you think, and it's really annoying because you can't see how many characters you actually deleted, and well, it's like writing blinded

**Comment by daniele-verducci** [comment]
For the "Simple editing" button to "jump" to top left of the screen you need to change the Direct Editing setting as explained by @fm-sys , that's why I couldn't reproduce it at first...
The other editing button (the Rich Text one) should already disappear while scrolling.
Now, with PR https://github.com/nextcloud/notes-android/pull/3328 , the "Simple editing" button should stay in the right place and should behave like the Direct Editing one, hiding when the text is scrolled. This should address this issue.

**Comment by daniele-verducci** [comment]
> Delete the last few characters; this will trigger the vertical scroll following the deletion of the last line, and the floating button will cover all the text from the previous line that you're trying to edit:

This edge case it's actually still there. I'll take a look at it!
